packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -533,11 +533,16 @@ abstract class FlutterCommand extends Command<void> {
 
     devices = devices.where((Device device) => device.isSupported()).toList();
     // If the user has not specified all devices and has multiple connected
-    // then filter then list by those supported in the current project. If
-    // this ends up with a single device we can proceed as normal.
+    // then filter then list by those supported in the current project and
+    // remove non-ephemeral device types. If this ends up with a single
+    // device we can proceed as normal.
     if (devices.length > 1 && !deviceManager.hasSpecifiedAllDevices && !deviceManager.hasSpecifiedDeviceId) {
       final FlutterProject flutterProject = FlutterProject.current();
       devices.removeWhere((Device device) => !device.isSupportedForProject(flutterProject));
+
+      if (devices.any((Device device) => device.ephemeral)) {
+        devices.removeWhere((Device device) => !device.ephemeral);
+      }
     }
 
```

Comment:
It would be nice to add a test that verifies this behavior including that the error messages shown are as expected.

Reply:
I refactored this logic into a top level function that is now tested